path_service_principal.go

```diff
@@ -184,11 +184,23 @@ func (b *azureSecretBackend) createSPSecret(ctx context.Context, s logical.Stora
 		return "", err
 	}
 
+	var pwExpiresAt time.Time
+	if password.EndDate != nil {
+		pwExpiresAt = password.EndDate.ToTime()
+	}
+
+	var keyID string
+	if password.KeyID != nil {
+		keyID = *password.KeyID
+	}
+
 	role.ApplicationID = appID
 	role.ApplicationObjectID = appObjID
 	role.ServicePrincipalID = spID
 	role.Credentials = &ClientCredentials{
-		Password: password,
+		KeyId:     keyID,
+		Password:  *password.SecretText,
```

Comment:
why do the other password fields have a nil check but not `SecretText`?

Reply:
The nil checks are probably over the top but I wanted to reduce the chances that currently working code paths would cause a panic. The secret text was already being derefed here so we know it is safe.